var/spack/repos/builtin/packages/groff/package.py

```diff
@@ -16,41 +16,62 @@ class Groff(AutotoolsPackage, GNUMirrorPackage):
     homepage = "https://www.gnu.org/software/groff/"
     gnu_mirror_path = "groff/groff-1.22.3.tar.gz"
 
+    version('1.22.4', sha256='e78e7b4cb7dec310849004fa88847c44701e8d133b5d4c13057d876c1bad0293')
+    version('1.22.3', sha256='3a48a9d6c97750bfbd535feeb5be0111db6406ddb7bb79fc680809cda6d828a5')
+
     # TODO: add html variant, spack doesn't have netpbm and its too
     # complicated for me to find out at this point in time.
     # See brew scripts for groff for guidance:
     # https://github.com/Homebrew/homebrew-core/blob/master/Formula/groff.rb
     # Seems troublesome...netpbm requires groff?
     variant('pdf', default=True, description='Build the `gropdf` executable.')
+    variant('x', default=False, description='Enable set of graphical options')
+    variant('uchardet', default=True,
+            description='Builds preconv with uchardet library for '
+                        'automatic file encoding detection')
```

Comment:
If I understand correctly, we miss:
```
conflicts('+uchardet', when='@:1.22.3')
```

Reply:
In 1.22.3 uchardet presence in not relevant.
Do you assert that we must raise a conflict if +ucharset is (with no efffect) set ?

Reply:
Yes, I'd like to prevent `@1.22.3+uchardet` to be a viable configuration since it doesn't make sense according to the recipe.

Reply:
But if we do that, we will not be able to compile 1.22.3 with the default (not specifying any variant) whereas its configure can accept an irrelevant (void) option.
The thing I would like to do, then, is to set the default uchardet variant to False when 1.22.3 and True when 1.22.4 : much like the parallel flag. But, for a variant, I have to search how one can do that, I have no idea.

Reply:
> But if we do that, we will not be able to compile 1.22.3 with the default (not specifying any variant) whereas its configure can accept an irrelevant (void) option.

This is true only for the original concretizer. The ASP-based solver should be able to work around conflicts.